modules/video/src/video-builder.ts

```diff
@@ -0,0 +1,75 @@
+const VIDEO_BUILDER_OPTIONS = {
+  width: 512,
+  height: 512,
+  framerate: 24,
+  bitrate: 200,
+  realtime: false
+};
+
+type VideoBuilderOptions = Partial<typeof VIDEO_BUILDER_OPTIONS>;
+
+function nextEvent(target, name): Promise<{data: any}> {
+  return new Promise((resolve) => {
+    target.addEventListener(name, resolve, {once: true});
+  });
+}
+
+export default class VideoBuilder {
+  worker: Worker = new Worker('node_modules/webm-wasm/dist/webm-worker.js');
+
+  static get properties() {
+    return {
+      id: 'webm',
+      name: 'WEBM',
+      extensions: ['webm'],
+      mimeTypes: ['video/webm'],
+      builder: VideoBuilder,
+      options: VIDEO_BUILDER_OPTIONS
+    };
+  }
+
+  async initialize(options: VideoBuilderOptions): Promise<void> {
```

Comment:
Maybe a JSDoc `/** */` comment on each method, it is nice when hovering in vscode if nothing else...

Reply:
Sounds good. Will add.